tests/unit/geo.tcl

```diff
@@ -456,4 +589,92 @@ start_server {tags {"geo"}} {
+            expr {srand($srand_seed)} ; # If you need a reproducible run
+            r del mypoints
+
+            geo_random_point search_lon search_lat
+            set width_m [expr {[randomInt 10000]+10}]
+            set height_m [expr {[randomInt 10000]+10}]
+            set lat_delta [geo_raddeg [expr {$height_m/2/6372797.560856}]]
+            set long_delta_top [geo_raddeg [expr {$width_m/2/6372797.560856/cos([geo_degrad [expr {$search_lat+$lat_delta}]])}]]
+            set long_delta_middle [geo_raddeg [expr {$width_m/2/6372797.560856/cos([geo_degrad $search_lat])}]]
+            set long_delta_bottom [geo_raddeg [expr {$width_m/2/6372797.560856/cos([geo_degrad [expr {$search_lat-$lat_delta}]])}]]
+
+            # Total of 8 points are generated, which are located at each vertex and the center of each side
+            set points(0) [expr {$search_lon-$long_delta_top}]
+            set points(1) [expr {$search_lat+$lat_delta}]
+            set points(2) [expr {$search_lon+$long_delta_top}]
+            set points(3) [expr {$search_lat+$lat_delta}]
+            set points(4) [expr {$search_lon+$long_delta_bottom}]
+            set points(5) [expr {$search_lat-$lat_delta}]
+            set points(6) [expr {$search_lon-$long_delta_bottom}]
+            set points(7) [expr {$search_lat-$lat_delta}]
+            set points(8) $search_lon
+            set points(9) [expr {$search_lat+$lat_delta}]
+            set points(10) $search_lon
+            set points(11) [expr {$search_lat-$lat_delta}]
+            set points(12) [expr {$search_lon+$long_delta_middle}]
+            set points(13) $search_lat
+            set points(14) [expr {$search_lon-$long_delta_middle}]
+            set points(15) $search_lat
+
+            lappend debuginfo "Search area: geosearch mypoints fromlonlat $search_lon $search_lat bybox $width_m $height_m m"
+            set tcl_result {}
+            for {set i 0} {$i < 16} {incr i 2} {
+                set x $points($i)
+                set y $points([expr {$i+1}])
+                r geoadd mypoints $x $y place:$i
+                lappend tcl_result "place:$i"
+                lappend debuginfo "geoadd mypoints $x $y place:$i"
+            }
+
+            set res2 [lsort $tcl_result]
+
+            # make the box larger by two meter in each direction to put the coordinate slightly inside the box.
+            set height_new [expr {$height_m+4}]
+            set width_new [expr {$width_m+4}]
+            set res [lsort [r geosearch mypoints fromlonlat $search_lon $search_lat bybox $width_new $height_new m]]
+            if {$res != $res2} {
+                set diff [compare_lists $res $res2]
+                lappend debuginfo "diff: $diff"
+                fail "place should be found, debuginfo: $debuginfo, height_new: $height_new width_new: $width_new"
+            }
+
+            # Only width reduced, the four corners and the midpoint on the left and right will not be searched
```

Comment:
did you mean
```suggestion
            # Only width reduced, the four corners and the midpoint on the left and right will not be found
```
doesn't this make the test possibly flaky? i.e. the two points that are found are right on the border, due to inaccuracy issues they may be found time times and not found other times?